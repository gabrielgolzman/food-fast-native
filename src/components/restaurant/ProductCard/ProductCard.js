import React from 'react';

import {
   ProductCardWhole,
   ProductCardPhoto,
   ProductTitle,
   ProductDescription,
   ProductPrice,
   Info,
} from './ProductCard.styles';

const ProductCard = ({ product, noElevation }) => {
   const { idMenuOption, optionName, description, unitPrice, photo } = product;
   return (
      <ProductCardWhole elevation={!noElevation ? 5 : 0}>
         <ProductCardPhoto
            alt={optionName}
            key={idMenuOption}
            source={{ uri: photo }}
         />
         <Info>
            <ProductTitle>{optionName}</ProductTitle>
            <ProductDescription>{description}</ProductDescription>
            <ProductPrice>${unitPrice}</ProductPrice>
         </Info>
      </ProductCardWhole>
   );
};

export default ProductCard;
