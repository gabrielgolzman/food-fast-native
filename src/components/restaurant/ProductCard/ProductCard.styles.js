import styled from 'styled-components/native';
import { Card } from 'react-native-paper';

export const ProductCardWhole = styled(Card)`
   background-color: ${(props) => props.theme.colors.ui.bg};
   margin-bottom: ${(props) => props.theme.space[3]};
`;

export const ProductCardPhoto = styled(Card.Cover)`
   padding: ${(props) => props.theme.space[2]};
   background-color: ${(props) => props.theme.colors.ui.bg};
`;

export const Info = styled.View`
   padding: ${(props) => props.theme.space[3]};
`;

export const ProductTitle = styled.Text`
   font-size: ${(props) => props.theme.fontSizes.h5};
   margin-bottom: ${(props) => props.theme.space[0]};
`;

export const ProductDescription = styled.Text`
   font-size: ${(props) => props.theme.fontSizes.caption};
   margin-bottom: ${(props) => props.theme.space[2]};
`;

export const ProductPrice = styled.Text`
   font-size: ${(props) => props.theme.fontSizes.title};
   font-weight: ${(props) => props.theme.fontWeights.bold};
`;

export const SectionEnd = styled.View`
   flex: 1;
   flex-direction: row;
   justify-content: flex-end;
`;

export const Icon = styled.Image`
   width: 15px;
   height: 15px;
`;
