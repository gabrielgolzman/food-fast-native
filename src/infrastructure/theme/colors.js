export const colors = {
   ui: {
      primary: '#F3A404',
      secondary: '#F1062E',
      bg: '#FFFFFF',
   },
};
