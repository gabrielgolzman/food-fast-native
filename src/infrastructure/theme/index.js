import { colors } from './colors';
import { space } from './spacing';
import { fontSizes, fontWeights } from './fonts';

export const theme = { colors, space, fontSizes, fontWeights };
