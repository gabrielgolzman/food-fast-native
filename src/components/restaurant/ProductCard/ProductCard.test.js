import React from 'react';
import { render } from '@testing-library/react-native';

import { theme } from '../../../infrastructure/theme/index';
import { ThemeProvider } from 'styled-components/native';
import ProductCard from './ProductCard';

it('renders the ProductCard correctly', () => {
   let mock = require('../../../../data/mock.json');
   const tree = render(
      <ThemeProvider theme={theme}>
         <ProductCard product={mock.menuOptions[0]} />
      </ThemeProvider>
   ).toJSON();
   expect(tree).toMatchSnapshot();
});
