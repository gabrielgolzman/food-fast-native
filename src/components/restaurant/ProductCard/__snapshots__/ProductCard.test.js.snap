// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the ProductCard correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderColor": "rgba(0, 0, 0, 0.12)",
      "borderRadius": 4,
      "elevation": 5,
      "marginBottom": 16,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 4,
        "width": 0,
      },
      "shadowOpacity": 0.24,
      "shadowRadius": 5,
    }
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flexGrow": 1,
        "flexShrink": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#eeeeee",
            "height": 195,
            "overflow": "hidden",
          },
          Object {
            "borderTopLeftRadius": 4,
            "borderTopRightRadius": 4,
          },
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
              "paddingBottom": 8,
              "paddingLeft": 8,
              "paddingRight": 8,
              "paddingTop": 8,
            },
          ],
        ]
      }
    >
      <Image
        alt="Café colombiano"
        siblings={
          Array [
            "Styled(withTheme(Card.Cover))",
            "Styled(View)",
          ]
        }
        source={
          Object {
            "uri": "https://www.mdzol.com/u/fotografias/m/2021/4/8/f850x638-1041350_1118839_5050.jpg",
          }
        }
        style={
          Array [
            Object {
              "flex": 1,
              "height": undefined,
              "justifyContent": "flex-end",
              "padding": 16,
              "width": undefined,
            },
            Object {
              "borderTopLeftRadius": 4,
              "borderTopRightRadius": 4,
            },
          ]
        }
      />
    </View>
    <View
      index={1}
      siblings={
        Array [
          "Styled(withTheme(Card.Cover))",
          "Styled(View)",
        ]
      }
      style={
        Array [
          Object {
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
          },
        ]
      }
      total={2}
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 24,
              "marginBottom": 0,
            },
          ]
        }
      >
        Café colombiano
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontSize": 12,
              "marginBottom": 8,
            },
          ]
        }
      >
        El mejor café colombiano que probarás jamás
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontSize": 20,
              "fontWeight": "700",
            },
          ]
        }
      >
        $
        150
      </Text>
    </View>
  </View>
</View>
`;
